File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int l=0;
        int n = height.size();
        int r = n-1;
        int ans=0;
        while(l<r) {
            ans = max(ans,min(height[l],height[r])*(r-l));
            if (height[l]<height[r]) {
                l+=1;
            } else {
                r-=1;
            }

        }
        return ans;
        
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        lIndex=0
        rIndex = len(height)-1
        ans=0
        while lIndex<rIndex:
            ans = max(ans,min(height[lIndex],height[rIndex])*(rIndex-lIndex))
            if height[lIndex]<height[rIndex]:
                 lIndex+=1
            else: rIndex-=1
        return ans
        